dna/clutter/clutter.js

```diff
@@ -42,15 +42,172 @@ function post(post) {
+    }      // On the DHT, puts a link on my hash to the new post
+
+
     debug("meta: "+JSON.stringify(getLink(me,"post",{Load:true})));
     debug(key);
     return key;                                  // Returns the hash key of the new post to the calling function
 }
 
+function searchHashTag(hashTag,post)
+{
+ for (var i = 0; i < hashTag.length; i++)
+ {
+//   debug("SEARCHING :");
+data=getHashTag(hashTag[i]);
+debug("Data"+data);
+
+  if (data != "") {
+    debug("HASHTAG matched IN DHT");
+             linkHashTags(hashTag[i],data,post); //follow)() changed
+         }
+         else {
+var a=hashTag[i];
+             debug("HASHTAG no match found");
+          //   debug(hashTag[i].value);
+        //  debug(a);
+      //    debug(typeof a);
+             //CREted createHashTag();
+          addHashTag(a);
+           debug("CREATING HASHTAG COMPLETE")
+           debug("ADDING NEW Hanstags:");
+           data=getHashTag(hashTag[i]);
+           debug("Data"+data);
+
+           if (data != "") {
+            debug("HASHTAG FOUND IN DHT");
+                     linkHashTags(hashTag[i],data,post); //follow)() changed
+                 }
+                 else {
+           var a=hashTag;
+                     debug("HASHTAG not found");
+                  //   debug(hashTag[i].value);
+                  debug(a);
+                  debug(typeof a);
+                     //CREted createHashTag();
+                  addHashTag(a);
+                   debug("CREATING Again HASHTAG COMPLETE")
+                 }
+
+       }
+
+
+       getPostsByTag(hashTag[i]); /**THIS METHORD CAN BE USE TO SEARCH THE HASH TAGS
+       <USED here to test if we can get the post to the recently entered hashTag>**/
+
+}
+}
+
+
+function linkHashTags(hashtag,hashTag_Address,post)
+{
+//  getLink(hashTag_Address,"tag_post",{Load:false};
+var hashtagHash = makeHash(hashtag);
+var Entrys = get(hashtagHash,{GetMask:HC.GetMask.Entry});
+var Default=   get(hashtagHash,{GetMask:HC.GetMask.Default});
+var EntryType=get(hashtagHash,{GetMask:HC.GetMask.EntryType});
+var All= get(hashtagHash,{GetMask:HC.GetMask.All});
+//HC.GetMask.Default/Entry/EntryType/Sources/All;
+/*debug("HASHTAG source links:: ");
+debug("Entrys"+JSON.stringify(Entrys));
+debug("Default"+JSON.stringify(Default));
+debug("EntryType"+JSON.stringify(EntryType));
+debug("All"+JSON.stringify(All));
+*/  var key = commit("tag_post",post);
+  commit("hashTag_links",{Links:[{Base:hashtagHash,Link:key,Tag:"tag_post"}]});
+  debug("hashTag_linked: "+JSON.stringify(getLink(hashtagHash,"tag_post",{Load:true})));
+
+}
+
+
+function getHashTag(hashtag)       //getAgent  //GIVES you the sources of the hashtag
+{
+    var directory = getDirectory();
+    var hashtagHash = makeHash(hashtag);
+    var sources = get(hashtagHash,{GetMask:HC.GetMask.Sources});
+debug("SOURCES:: "+ sources);
+for(var i=0;i<sources.length;i++)
+debug("SOURCES::"+i+"= "+sources[i]);
+    if (isErr(sources)) {sources = [];}
```

Comment:
Since this next sequence is now used in more than one place (getAgent) it should be refactored out into a more meaningful function call